backend/connector-integration/src/connectors/archipel/transformers.rs

```
@@ -0,0 +1,1586 @@
+                ArchipelPaymentFlow::Cancel => Self::Voided,
+            },
+            ArchipelPaymentStatus::Failed => match flow {
+                ArchipelPaymentFlow::Authorize | ArchipelPaymentFlow::Pay => {
+                    Self::AuthorizationFailed
+                }
+                ArchipelPaymentFlow::Verify => Self::AuthenticationFailed,
+                ArchipelPaymentFlow::Capture => Self::CaptureFailed,
+                ArchipelPaymentFlow::Cancel => Self::VoidFailed,
+            },
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ArchipelOrderResponse {
+    id: String,
+    amount: Option<MinorUnit>,
+    currency: Option<Currency>,
+    captured_amount: Option<MinorUnit>,
+    authorized_amount: Option<MinorUnit>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct ArchipelErrorMessage {
+    pub code: String,
+    pub description: Option<String>,
+}
+
+impl Default for ArchipelErrorMessage {
+    fn default() -> Self {
+        Self {
+            code: consts::NO_ERROR_CODE.to_string(),
+            description: Some(consts::NO_ERROR_MESSAGE.to_string()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct ArchipelErrorMessageWithHttpCode {
+    pub error_message: ArchipelErrorMessage,
+    pub http_code: u16,
+}
+
+impl ArchipelErrorMessageWithHttpCode {
+    pub fn new(error_message: ArchipelErrorMessage, http_code: u16) -> Self {
+        Self {
+            error_message,
+            http_code,
+        }
+    }
+
+    pub fn from_response(response: &[u8], http_code: u16) -> Self {
+        let error_message = if response.is_empty() {
+            ArchipelErrorMessage::default()
+        } else {
+            serde_json::from_slice::<ArchipelErrorMessage>(response)
+                .unwrap_or_else(|_| ArchipelErrorMessage::default())
```

Don't discard the actual error, instead log it.

---

ok